### PR TITLE
[bitnami/minio] add prometheus rule

### PR DESCRIPTION
**Pull request by lusson-luo**
Signed-off-by: lusson <lusson@foxmail.com>

### Description of the change

minio charts add prometheusRule config

### Benefits

minio charts can add prometheusRule

### Possible drawbacks

no

### Applicable issues

no

### Additional information

no

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by javsalgar** [comment]
Hi! Thanks for the PR! Could you fix the merge and DCO issues?

**Comment by lusson-luo** [comment]
> Hi! Thanks for the PR! Could you fix the merge and DCO issues?

ok

File: bitnami/minio/templates/prometheusrule.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    {{- if .Values.metrics.prometheusRule.additionalLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.metrics.prometheusRule.additionalLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  groups:
+  - name: {{ include "common.names.fullname" . }}
+    rules: {{- include "common.tplvalues.render" ( dict "value" .Values.metrics.prometheusRule.rules "context" $ ) | nindent 6 }}
```
Comment (FraPazGal) [inline bitnami/minio/templates/prometheusrule.yaml:25]:
```suggestion
    rules: {{- include "common.tplvalues.render" ( dict "value" .Values.metrics.prometheusRule.rules "context" $ ) | nindent 8 }}
```

File: bitnami/minio/templates/prometheusrule.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{{- if and .Values.metrics.enabled .Values.metrics.prometheusRule.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: {{ include "common.names.fullname" . }}
+  {{- if .Values.metrics.prometheusRule.namespace }}
+  namespace: {{ .Values.metrics.prometheusRule.namespace }}
+  {{- else }}
+  namespace: {{ .Release.Namespace }}
+  {{- end }}
```
Comment (FraPazGal) [inline bitnami/minio/templates/prometheusrule.yaml:10]:
```suggestion
  namespace: {{ default .Release.Namespace .Values.metrics.prometheusRule.namespace | quote }}
```

File: bitnami/minio/Chart.yaml [hunk 3]
```diff
@@ -25,4 +25,4 @@ name: minio
 sources:
   - https://github.com/bitnami/bitnami-docker-minio
   - https://min.io
-version: 11.5.1
+version: 11.5.2
```
Comment (FraPazGal) [inline bitnami/minio/Chart.yaml:28]: As we are adding a new feature, we should bump the chart's version to a new minor.
```suggestion
version: 11.6.0
```

**Review by FraPazGal — CHANGES_REQUESTED** [review]
Hello @lusson-luo, I added a few comments to your PR. Could you check them out?

File: bitnami/minio/templates/prometheusrule.yaml (hunk 1 above)
Comment (lusson-luo) [reply bitnami/minio/templates/prometheusrule.yaml:25]: I think it's 6, cause rules has 4 space, it has 2 spaces more than rules, so it's 6 space

Comment (lusson-luo) [reply bitnami/minio/templates/prometheusrule.yaml:21]: @FraPazGal what do you think about this?

**Review by FraPazGal — APPROVED** [review]
LGTM

File: bitnami/minio/templates/prometheusrule.yaml (hunk 1 above)
Comment (FraPazGal) [reply bitnami/minio/templates/prometheusrule.yaml:21]: I think you are right @lusson-luo. We can mark this as resolved.

**Pull request merged**